perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1697,7 +1697,9 @@ public static boolean isMfaAuthorizedForAttribute(PerunSession sess, AttributeDe
 			return principalMfa || updatePrincipalMfa(sess);
 		}
 
-		return principalMfa || !isAnyObjectMfaCritical(sess, objects) || updatePrincipalMfa(sess);
+		boolean globallyCriticalAction  = ((PerunBl) sess.getPerun()).getAttributesManagerBl().isAttributeActionGloballyCritical(sess, attrDef, actionType);
+
+		return principalMfa || !isAnyObjectMfaCritical(sess, objects) || !globallyCriticalAction || updatePrincipalMfa(sess);
```

Comment:
If I understand correctly, we want to completely skip the critical object check when the action is globally critical, no? This way `!isAnyObjectMfaCritical(sess, objects)` would return true if no objects were critical and we would not check global criticality.

Reply:
You're right. I think that the improved condition should resolve this problem.